website/docs/docs/build/unit-tests.md

```diff
@@ -0,0 +1,139 @@
+        )
+        = true
+        and accepted_email_domains.tld is not null,
+        false) as is_valid_email_address
+    from customers
+		left join accepted_email_domains
+        on customers.email_top_level_domain = lower(accepted_email_domains.tld)
+
+)
+
+select * from check_valid_emails
+```
+
+This type of logic can be challenging to validate. Let’s add a unit test to this model to ensure our `is_valid_email_address` logic captures all known edge cases: emails without `.`, emails without `@`, and emails from invalid domains.
+
+```yaml
+unit_tests:
+  - name: test_is_valid_email_address
+    description: "Check my is_valid_email_address logic captures all known edge cases - emails without ., emails without @, and emails from invalid domains."
+    model: dim_customers
+    given:
+      - input: ref('stg_customers')
+        rows:
+          - {customer_id: 1, email: cool@example.com, email_top_level_domain: example.com}
+          - {customer_id: 2, email: cool@unknown.com, email_top_level_domain: unknown.com}
+          - {customer_id: 3, email: badgmail.com, email_top_level_domain: gmail.com}
+          - {customer_id: 4, email: missingdot@gmailcom, email_top_level_domain: gmail.com}
+      - input: ref('top_level_email_domains')
+        rows:
+          - {tld: example.com}
+          - {tld: gmail.com}
+    expect:
+      rows:
+        - {customer_id: 1, is_valid_email_address: true}
+        - {customer_id: 2, is_valid_email_address: false}
+        - {customer_id: 3, is_valid_email_address: false}
+        - {customer_id: 4, is_valid_email_address: false}
+
+```
+
+The above example defines the mock data using the inline `dict` format, but there are a handful of different options for how you format your mock data. 
+
+You’ll notice that we _only_ had to define the mock data for the columns we care about. This enables you to write succinct and _specific_ unit tests.
+
+:::note
+
+The direct parents of the model that you’re unit testing (in this example `stg_customers` and `top_level_email_domains`) need to exist in the warehouse before you’re able to execute the unit test.
+
+Use the `—-empty` flag to build an empty version of those models to save warehouse spend. 
+
+```bash
+
+`dbt run —-select "stg_customers top_level_email_domains" --empty`.
+
+```
+
+Alternatively, use `dbt build` to, in lineage order:
+
+- Run the unit tests on your model.
+- Materialize your model in the warehouse.
+- Run the data tests on your model.
+
+:::
+
+Now we’re ready to run this unit test! We have a couple of options for commands depending on how specific we want to be: 
+
+- `dbt test —-select dim_customers` runs _all_ of the tests on `dim_customers`.
+- `dbt test —-select "dim_customers,test_type:unit"` runs all of the _unit_ tests on `dim_customers`.
+- `dbt test —-select test_is_valid_email_address` runs the test named `test_is_valid_email_address`.
+
+[add screenshot]
```

Comment:
@matthewshaver would you prefer these screenshots come from core or cloud IDE? I can also just copy/paste the logs if that's better?

Reply:
@graciegoheen I like the idea of copy/paste the logs so we don't have to commit to one product or the other